src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -396,9 +397,9 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
     )
 
     if no_wait:
-        return sdk_no_wait(no_wait, client.create, resource_group_name, cluster_name, create_params)
+        return sdk_no_wait(no_wait, client.begin_create, resource_group_name, cluster_name, create_params)
```

Comment:
This issue is not related to track 2 migration:
We just need `return sdk_no_wait(no_wait, client.begin_create, resource_group_name, cluster_name, create_params)`, no need to check `if no_wait` as `sdk_no_wait()` will handle this logic based on the value of `no_wait`.

Reply:
removed the duplicate check logic, thanks